src/util/belt.rs

```diff
@@ -133,18 +132,17 @@ impl StagingBelt {
             self.free_chunks.push(chunk);
         }
 
-        let sender_template = &self.sender;
-        join_all(self.closed_chunks.drain(..).map(|chunk| {
-            let sender = sender_template.clone();
-            chunk
-                .buffer
-                .slice(..)
-                .map_async(MapMode::Write)
-                .inspect(move |_| {
-                    // The only possible error is the other side disconnecting, which is fine
-                    let _ = sender.send(chunk);
-                })
-        }))
-        .map(|_| ())
+        let sender = self.sender.clone();
+        let chunks = self.closed_chunks.drain(..).collect::<Vec<_>>();
+
+        async move {
+            for chunk in chunks {
+                // The result is ignored
+                chunk.buffer.slice(..).map_async(MapMode::Write).await.ok();
```

Comment:
If I'm understand correctly it seems like this would wait for every chunk in order. Instead I think we'd want to start running each future (e.g. `map_async`) and `await` them in parallel/poll them all in a loop (but not blocking on any of them), which is what `join_all` seems to do (https://docs.rs/futures-util-preview/0.3.0-alpha.19/src/futures_util/future/join_all.rs.html#115-122)

Reply:
@grovesNL Glad you caught that. The latest commit rewrites this one to return a future that polls its inner futures similarly to `join_all`.

I've only tested it with the `skybox` demo, but it is properly allowing chunks to be reused in the `StagingBelt`, and I don't see any obvious flaws with the implementation.